lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -3,46 +3,79 @@
 import com.mapzen.android.lost.api.FusedLocationProviderApi;
 import com.mapzen.android.lost.api.LocationListener;
 import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LostApiClient;
 
 import android.app.PendingIntent;
+import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
+import android.content.ServiceConnection;
 import android.location.Location;
+import android.os.IBinder;
 import android.os.Looper;
 import android.util.Log;
 
 import java.io.File;
-import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Implementation of the {@link FusedLocationProviderApi}.
  */
 public class FusedLocationProviderApiImpl
-    implements FusedLocationProviderApi, LocationEngine.Callback {
+    implements FusedLocationProviderApi {
 
   private static final String TAG = FusedLocationProviderApiImpl.class.getSimpleName();
 
   private final Context context;
-  private boolean mockMode;
-  private LocationEngine locationEngine;
-  private Map<LocationListener, LocationRequest> listenerToRequest;
-  private Map<PendingIntent, LocationRequest> intentToRequest;
+  private FusedLocationProviderService service;
+
+  private final ServiceConnection serviceConnection = new ServiceConnection() {
+    @Override public void onServiceConnected(ComponentName name, IBinder binder) {
+      FusedLocationProviderService.FusedLocationProviderBinder fusedBinder =
+          (FusedLocationProviderService.FusedLocationProviderBinder) binder;
+      if (fusedBinder != null) {
+        service = fusedBinder.getService();
+      }
+
+      if (connectionCallbacks != null) {
+        connectionCallbacks.onConnected();
+      }
+      Log.d(TAG, "[onServiceConnected]");
+    }
+
+    @Override public void onServiceDisconnected(ComponentName name) {
+      if (connectionCallbacks != null) {
+        connectionCallbacks.onConnectionSuspended();
+      }
+      Log.d(TAG, "[onServiceDisconnected]");
+    }
+  };
+
+  LostApiClient.ConnectionCallbacks connectionCallbacks;
 
   public FusedLocationProviderApiImpl(Context context) {
     this.context = context;
-    locationEngine = new FusionEngine(context, this);
-    listenerToRequest = new HashMap<>();
-    intentToRequest = new HashMap<>();
+  }
+
+  public void connect(LostApiClient.ConnectionCallbacks callbacks) {
+    Intent intent = new Intent(context, FusedLocationProviderService.class);
+    context.startService(intent);
+
+    connectionCallbacks = callbacks;
+    intent = new Intent(context, FusedLocationProviderService.class);
+    context.bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+  }
+
+  public void disconnect() {
+    context.unbindService(serviceConnection);
```

Comment:
Ha ok I didn't think about using the application context. This seems fine if not the preferred behavior. It makes sense that if you use an activity as the context the lifecycle of the connection would be bound to the lifecycle of the activity and if you use the application context it would not. Is this how play services behaves too?